### ISSUE TITLE
Memory leak after session recovery

### ISSUE THREAD

**Issue by romainmoreau** [Lightstreamer/Lightstreamer-lib-client-java#1]
Hi,

In some cases, after the recovery of a session, the value of the following field always remains `true`:

https://github.com/Lightstreamer/Lightstreamer-lib-client-java/blob/c0a5ca912c967fc548b4ce6e818c15b14712aa12/src/com/lightstreamer/client/protocol/TextProtocol.java#L108

It ends up with a memory leak because the `ArrayList` `capturedMessages` of the same class keeps growing.

This issue happens only sometimes and some other times the recovery works fine, without changing anything in the code.

**Comment by acarioni** [comment]
Thanks for reporting the issue.

If possible, can you provide a log showing a scenario triggering the memory leak?

To enable logging, simply add these lines to your application

```
SystemOutLogProvider loggerProvider = new SystemOutLogProvider();
LightstreamerClient.setLoggerProvider (loggerProvider);
```

and these libraries in dependencies

```
implementation ("com.lightstreamer: ls-log-adapter-java: 1.0.2")
implementation ("com.lightstreamer: java-system-out-log: 1.0.2")
```

If the log is too big, please send it to _support@lightstreamer.com_.

**Comment by romainmoreau** [comment]
For privacy reasons, I sent you the log by mail.

**Comment by acarioni** [comment]
Fixed by the versions 4.3.8 for JavaSE and 4.2.6 for Android.
